include/deal.II/optimization/line_minimization.h

```diff
@@ -0,0 +1,620 @@
+   * @p x_low @p x_hi @p x_rec[0] and derivatives at two points @p g_low and @p g_hi. The returned point
+   * should be within the bounds @p bounds .
+   *
+   * This function will first try the cubic fit (see cubic_fit ). If it's
+   * unsuccessfull or not
+   * within the provided @p bounds the quadratic fit will be performed (see quadratic_fit ). The function will
+   * fallback to bisection if quadratic fit fails.
+   */
+  template <typename NumberType>
+  NumberType
+  poly_fit(const NumberType                        x_low,
+           const NumberType                        f_low,
+           const NumberType                        g_low,
+           const NumberType                        x_hi,
+           const NumberType                        f_hi,
+           const NumberType                        g_hi,
+           const FiniteSizeHistory<NumberType> &   x_rec,
+           const FiniteSizeHistory<NumberType> &   f_rec,
+           const FiniteSizeHistory<NumberType> &   g_rec,
+           const std::pair<NumberType, NumberType> bounds);
+
+  /**
+   * Same as above but doing cubic fit with three points (see
+   * cubic_fit_three_points ).
+   */
+  template <typename NumberType>
+  NumberType
+  poly_fit_three_points(const NumberType                        x_low,
+                        const NumberType                        f_low,
+                        const NumberType                        g_low,
+                        const NumberType                        x_hi,
+                        const NumberType                        f_hi,
+                        const NumberType                        g_hi,
+                        const FiniteSizeHistory<NumberType> &   x_rec,
+                        const FiniteSizeHistory<NumberType> &   f_rec,
+                        const FiniteSizeHistory<NumberType> &   g_rec,
+                        const std::pair<NumberType, NumberType> bounds);
+
+
+  /**
+   * Perform a line search in $(0,max]$ with strong Wolfe conditions
+   * \f[
+   * f(\alpha) \le f(0) + \alpha \mu f'(0) \\
+   * |f'(\alpha)| \le \eta |f'(0)|
+   * \f]
+   * using one dimensional
+   * functions @p func and a function @p interpolate to choose a new point
+   * from the interval based on the function values and derivatives at its ends.
+   * @p a1 is a trial estimate of the first step.
+   * Interpolation can be done using poly_fit or poly_fit_three_points .
+   *
+   * The function implements Algorithms
+   * 2.6.2 and 2.6.4 on pages 34-35 in Fletcher, 2013, Practical methods of
+   * optimization. these are minor variations of the Algorithm 3.5 and 3.6 on
+   * pages 60-61 in Nocedal and Wright, Numerical optimization.
+   * It consists of a bracketing phase and a zoom phase, where @p interpolate is used.
+   *
+   * The function returns the step size and the number of times function @p func was called.
+   *
+   * @param func a one dimensional function which returns value and derivative
```

Comment:
doxygen renders this as a separate sentence, so capitalize the "A" in "a one dimensional...". same for all arguments below